src/main/java/org/apache/commons/io/output/FileWriterWithEncoding.java

```diff
@@ -227,7 +227,12 @@ private static Writer initWriter(final File file, final Object encoding, final b
         OutputStream stream = null;
         final boolean fileExistedAlready = file.exists();
         try {
-            stream = Files.newOutputStream(file.toPath(),  append ? StandardOpenOption.APPEND : StandardOpenOption.CREATE);
+            if (append) {
```

Comment:
Not sure why the fix has to be exploded into so much code when all you need is an additional argument to the JRE API. Please see git master for a much simpler fix.

Reply:
@garydgregory Thanks for the warm welcome as someone trying to help.
Have done another PR that shows, that your optimized version of the fix now breaks the ctor with the append option set to true.